helm-charts/basehub/values.yaml

```diff
@@ -580,27 +580,35 @@ jupyterhub:
         memory: 2Gi
     extraConfig:
       01-custom-theme: |
+        # adds a JupyterHub template path and updates template variables
+
         from z2jh import get_config
         c.JupyterHub.template_paths.insert(0,'/usr/local/share/jupyterhub/custom_templates')
         c.JupyterHub.template_vars.update({
             'custom': get_config('custom.homepage.templateVars')
         })
       02-custom-admin: |
+        # adjusts the spawner to add config specific to admin users
+
         from z2jh import get_config
         from kubespawner import KubeSpawner
         from jupyterhub_configurator.mixins import ConfiguratorSpawnerMixin
 
         class CustomSpawner(ConfiguratorSpawnerMixin, KubeSpawner):
             def start(self, *args, **kwargs):
                 custom_admin = get_config('custom.singleuserAdmin', {})
-                if custom_admin and self.user.admin:
-                    extra_init_containers = custom_admin.get('initContainers', [])
-                    extra_volume_mounts = custom_admin.get('extraVolumeMounts', [])
-                    extra_env = custom_admin.get('extraEnv', {})
-
-                    self.init_containers += [container for container in extra_init_containers if container not in self.init_containers]
-                    self.volume_mounts += [volume for volume in extra_volume_mounts if volume not in self.volume_mounts]
-                    self.environment.update(extra_env)
+                if not (self.user.admin and custom_admin):
+                    return super().start(*args, **kwargs)
```

Comment:
This is opinionated refactoring to have us avoid nesting into if statements and make the code a bit flatter.